Implement per store Kiyoh reviews

jesse-deboer opened the pull request:
- Added a command and service to retrieve reviews
- Changed the retrieval of reviews to get the reviews per storeview

Commit:
Implement per store Kiyoh reviews

- Added a command and service to retrieve reviews
- Changed the retrieval of reviews to get the reviews per storeview

janmartenjongerius commented on src/Console/Command/RetrieveReviews.php:
```diff
@@ -0,0 +1,46 @@
+        $this->service = $service;
+    }
+
+    public function configure()
```
Add return type `void`.

janmartenjongerius commented on src/Console/Command/RetrieveReviews.php:
```diff
@@ -0,0 +1,46 @@
+        try {
+            $this->service->execute();
+        } catch (LocalizedException $e) {
+            $output->write($e->getMessage());
```
You want to use the error output for errors.

```php
$errorOutput = $output instanceof ConsoleOutputInterface
    ? $output->getErrorOutput()
    : $output;

$errorOutput->write($e->getMessage());
```

Alternatively, in this case you don't want the program to proceed, so you're better off removing the `try-catch` altogether and let the exception handler take care of this. That way you also see a stack trace when you use an output with additional verbosity.

janmartenjongerius commented on src/Service/RetrieveKiyohReviews.php:
```diff
@@ -0,0 +1,124 @@
+            $this->curl->addHeader('X-Publication-Api-Token', $this->config->getApiKey((int) $store->getId()));
+            $this->curl->get(sprintf(Config::PUBLICATION_URL, $this->config->getLocationId((int) $store->getId())));
+
+            $output = $this->serializer->unserialize($this->curl->getBody());
```
You should really wrap this in a `try-catch` and let the catch also populate the `$errors` array., then `continue`.

janmartenjongerius commented on src/Service/RetrieveKiyohReviews.php:
```diff
@@ -0,0 +1,124 @@
+                    $output['errorCode'],
+                    $output['detailedError'][0]['message'] ?? null
+                );
+            }
```
I would add a `continue` statement after this, unless you really want to store the faulty response data in the database.
I think it's better to keep relying on the old state than to break it in the front-end.

janmartenjongerius reviewed (changes requested):
Looking really great so far. Just some minor adjustments :)

Commit:
Changes based on feedback

- Added return types
- Added a try catch and continue to continue to another store if there are errors
- Changed error output to check for instance of error output

Commit:
Removed try catch in execute function
